sympy/solvers/solveset.py

```diff
@@ -340,7 +341,16 @@ def _invert_complex(f, g_ys, symbol):
                                imageset(Lambda(n, f.inverse()(n)), g_ys), symbol)
 
     if isinstance(f, exp) or (f.is_Pow and f.base == S.Exp1):
-        if isinstance(g_ys, FiniteSet):
+        if any(e for e in Add.make_args(f) if len(e.atoms(exp)) == 2):
+            # can solve upto `(d*exp(exp(a*x+b)) + c)` format and `a` should be present
```

Comment:
These conditions seem like they would match other cases than the form shown:
```python
In [11]: f = exp(sin(exp(x)))

In [12]: f
Out[12]: 
    ⎛ x⎞
 sin⎝ℯ ⎠
ℯ       

In [13]: isinstance(f, exp)
Out[13]: True

In [14]: any(e for e in Add.make_args(f) if len(e.atoms(exp)) == 2)
Out[14]: True
```
The check should be more precise